### PR TITLE
[onnxifi] Support more operations.

### PR DESCRIPTION
**Pull request by rdzhabarov**
Cover all operations from resnet50 in the `compatibility` call.
cc: @yinghai @Maratyszcza

### CODE REVIEW

**Comment by yinghai** [comment]
@rdzhabarov  Check an alternative solution that covers all the ops that Glow supports now for ONNX. Neither solution is a complete check as when we convert one caffe2 op, it may result in multiple onnx nodes and one onnx node may translate to multiple Glow node. The translate logic is pretty specific. My solution is just check whether the type of op is supported.

**Comment by rdzhabarov** [comment]
We also need to check whether specific backend supports the operation, that's what's done here.

We could support parsing of specific ONNX operation but it does not guarantee that the backend can execute it.

**Comment by yinghai** [comment]
Then we need to address this: 
> Neither solution is a complete check as when we convert one caffe2 op, it may result in multiple onnx nodes and one onnx node may translate to multiple Glow node.

**Comment by rdzhabarov** [comment]
> We do need to care about multiple ONNX nodes, because one C2 node will be translated into multiple ONNX nodes

Oh, I though compatibility check is always for one entity/node
cc: @Maratyszcza

**Comment by yinghai** [comment]
Yeah, I think we might need to adjust the wording of it as it might not be the case.

File: lib/Importer/ONNXIFILoader.cpp [hunk 1]
```diff
@@ -135,36 +135,71 @@ std::unique_ptr<ModelLoader> ModelLoader::parse(
...
-  std::unique_ptr<std::pair<Kinded::Kind, ElemKind>> result;
   const std::string &operation = graph.node(0).op_type();
 
   // Quantized and non-quantized operations are handled by
   // different ONNX operators, for now only handle fp32.
   // TODO: Add more operators.
-  if (operation == "conv") {
-    result.reset(new std::pair<Kinded::Kind, ElemKind>(
+  if (operation == "BatchNormalization") {
+    result.push_back(std::pair<Kinded::Kind, ElemKind>(
```
Comment (yinghai) [inline lib/Importer/ONNXIFILoader.cpp]: nit: you can just do `result.emplace_back(Kinded::Kind::BatchNormalizationNodeKind, ElemKind::FloatTy)`

File: lib/Importer/ONNXIFILoader.cpp [hunk 2]
```diff
@@ -135,36 +135,54 @@ std::unique_ptr<ModelLoader> ModelLoader::parse(
...
-  std::unique_ptr<std::pair<Kinded::Kind, ElemKind>> result;
   const std::string &operation = graph.node(0).op_type();
 
   // Quantized and non-quantized operations are handled by
   // different ONNX operators, for now only handle fp32.
   // TODO: Add more operators.
-  if (operation == "conv") {
-    result.reset(new std::pair<Kinded::Kind, ElemKind>(
-        Kinded::Kind::ConvolutionNodeKind, ElemKind::FloatTy));
+  if (operation == "BatchNormalization") {
```
Comment (rdzhabarov) [inline lib/Importer/ONNXIFILoader.cpp]: talked to @opti-mix. Will change that to StringSwitch and add some macro for  lines like result.emplace_back(Kinded::Kind::BatchNormalizationNodeKind, ElemKind::FloatTy);

**Pull request merged**